staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -58,6 +66,9 @@ type AvailableConditionController struct {
 	endpointsLister v1listers.EndpointsLister
 	endpointsSynced cache.InformerSynced
 
+	enableAggregatorRouting bool
+	serviceResolver         ServiceResolver
```

Comment:
why do you need to wire both of these? serviceResolver should abstract away the enable flag.